packages/teleport/src/services/audit/types.ts

```diff
@@ -249,7 +286,258 @@ export type RawEvents = {
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_OPENDIR]: RawEvent<
+    typeof eventCodes.SFTP_OPENDIR,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_OPENDIR_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_OPENDIR_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_READDIR]: RawEvent<
+    typeof eventCodes.SFTP_READDIR,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_READDIR_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_READDIR_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_REMOVE]: RawEvent<
+    typeof eventCodes.SFTP_REMOVE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_REMOVE_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_REMOVE_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_MKDIR]: RawEvent<
+    typeof eventCodes.SFTP_MKDIR,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_MKDIR_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_MKDIR_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_RMDIR]: RawEvent<
+    typeof eventCodes.SFTP_RMDIR,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_RMDIR_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_RMDIR_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_REALPATH]: RawEvent<
+    typeof eventCodes.SFTP_REALPATH,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_REALPATH_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_REALPATH_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_STAT]: RawEvent<
+    typeof eventCodes.SFTP_STAT,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_STAT_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_STAT_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_RENAME]: RawEvent<
+    typeof eventCodes.SFTP_RENAME,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_RENAME_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_RENAME_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_READLINK]: RawEvent<
+    typeof eventCodes.SFTP_READLINK,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_READLINK_FAILURE]: RawEvent<
+    typeof eventCodes.SFTP_READLINK_FAILURE,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
+  >;
+  [eventCodes.SFTP_SYMLINK]: RawEvent<
+    typeof eventCodes.SFTP_SYMLINK,
+    {
+      path: string;
+      ['addr.local']: string;
+    }
```

Comment:
since `sftpXXX` has common fields we can create a type just for this event so we don't define these repeatedly
```
type RawEventSftp<T extends EventCode> = RawEvent<
  T,
  {
    path: string;
    ['addr.local']: string;
  }
>;
```